angularSTEP/src/app/upload-recipe/upload-recipe.component.ts

```diff
@@ -92,17 +92,28 @@ export class UploadRecipeComponent {
   }
 
   onSubmit() {
-    this.db.collection('recipes').add({
-      recipeName: this.autoCapitalizeName(this.recipeForm.value.name),
-      description: this.autoCapitalizeFirst(this.recipeForm.value.description),
-      ingredients: this.ingredients,
-      tools: this.tools,
-      instructions: this.instructions,
-      extraInfo: this.autoCapitalizeFirst(this.recipeForm.value.extraInfo),
-      difficulty: this.recipeForm.value.difficulty,
-      timestamp: Date.now(),
-    });
-    
+    if (this.autoCapitalizeFirst(this.recipeForm.value.extraInfo)) {
+      this.db.collection('recipes').add({
+        recipeName: this.autoCapitalizeName(this.recipeForm.value.name),
+        description: this.autoCapitalizeFirst(this.recipeForm.value.description),
+        ingredients: this.ingredients,
+        tools: this.tools,
+        instructions: this.instructions,
+        extraInfo: this.autoCapitalizeFirst(this.recipeForm.value.extraInfo),
+        difficulty: this.recipeForm.value.difficulty,
+        timestamp: Date.now(),
+      });
+    } else {
+      this.db.collection('recipes').add({
+        recipeName: this.autoCapitalizeName(this.recipeForm.value.name),
```

Comment:
Lol again, here, just to give you an idea of what I was saying up there. You then would be able to make a class with all of these things and add it to the database.